internal/appsec/rules_manager.go

```diff
@@ -99,15 +99,15 @@ func newRulesManager(rules []byte) (*rulesManager, error) {
 	}, nil
 }
 
-func (r *rulesManager) clone() *rulesManager {
+func (r *rulesManager) clone() rulesManager {
```

Comment:
Seems to me that this only makes the call more costly 🤔

Reply:
I benchmarked this and the benchmark actually seems to indicate that it actually makes the call cheaper (saves 1 allocation worth 480 bytes, and about 100ns/op)!

```console
$ go test -benchmem -run=^$ -tags appsec -bench ^BenchmarkRulesManagerClone$ gopkg.in/DataDog/dd-trace-go.v1/internal/appsec
goos: darwin
goarch: arm64
pkg: gopkg.in/DataDog/dd-trace-go.v1/internal/appsec
BenchmarkRulesManagerClone/struct-10             1436452               783.1 ns/op          1920 B/op          8 allocs/op
BenchmarkRulesManagerClone/pointer-10            1390646               887.2 ns/op          2400 B/op          9 allocs/op
```

The benchmark follows (in case you want to dispute my methodology):
```go
package appsec

import (
	"testing"

	"github.com/stretchr/testify/require"
)

var pointer *rulesManager

func BenchmarkRulesManagerClone(b *testing.B) {
	original, err := newRulesManager(nil)
	require.NoError(b, err)

	original.edits = map[string]rulesFragment{
		"overrides1/path": rulesFragment{
			Overrides: []interface{}{
				testRulesOverrideEntry{
					ID:      "crs-941-290",
					Enabled: false,
				},
				testRulesOverrideEntry{
					ID:      "crs-930-100",
					Enabled: false,
				},
			},
		},
		"overrides2/path": rulesFragment{
			Overrides: []interface{}{
				testRulesOverrideEntry{
					ID:      "crs-941-300",
					Enabled: false,
				},
				testRulesOverrideEntry{
					Enabled: false,
					ID:      "crs-921-160",
				},
			},
		},
	}

	b.Run("struct", func(b *testing.B) {
		b.StopTimer()
		b.ReportAllocs()

		r := *original
		b.StartTimer()
		for i := 0; i < b.N; i++ {
			r = r.clone()
		}
		b.StopTimer()
		pointer = &r
	})

	b.Run("pointer", func(b *testing.B) {
		b.StopTimer()
		b.ReportAllocs()

		var r *rulesManager = original
		b.StartTimer()
		for i := 0; i < b.N; i++ {
			r = r.clonePtr()
		}
		b.StopTimer()
		pointer = r
	})
}

func (r *rulesManager) clonePtr() *rulesManager {
	var clone rulesManager
	clone.edits = make(map[string]rulesFragment, len(r.edits))
	for k, v := range r.edits {
		clone.edits[k] = v
	}
	clone.base = r.base.clone()
	clone.latest = r.latest.clone()
	return &clone
}
```